x-pack/plugins/reporting/server/lib/validate/validate_browser.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as puppeteer from 'puppeteer-core';
+import { getAbsoluteUrlFactory } from '../../../common/get_absolute_url';
+import { KbnServer, Logger } from '../../../types';
+import { CHROMIUM } from '../../browsers/browser_types';
+
+const API_STATS_ENDPOINT = '/api/stats';
+
+/*
+ * Validate the Reporting headless browser can launch, and that it can connect
+ * to the locally running Kibana instance.
+ */
+export const validateBrowser = async (server: KbnServer, browserFactory: any, logger: Logger) => {
+  if (browserFactory.type === CHROMIUM) {
+    return browserFactory
+      .test(
+        {
+          viewport: { width: 800, height: 600 },
+        },
+        logger
+      )
+      .then(async (browser: puppeteer.Browser | null) => {
+        if (browser && browser.newPage) {
+          const getAbsoluteUrl = getAbsoluteUrlFactory(server);
+          const page = await browser.newPage();
+          const url = getAbsoluteUrl({ path: API_STATS_ENDPOINT });
+          logger.debug(`Opening page ${url}`);
+          await page.goto(url, { waitUntil: 'networkidle0' }); // Look for JSON response
```

Comment:
Other than the various moving files around and changing a few interfaces, this is the "key" change

Reply:
Sweet. I was just thinking it'd be a good idea to expand this test and make sure we can navigate. Glad you beat me to it!